File: frontend/src/services/llmUsage.ts
import api from './api'

export interface UsageSummaryRow {
  username: string
  display_name: string
  calls: number
  prompt_tokens: number
  completion_tokens: number
  total_tokens: number
  cost: number
  last_used: string
}

export interface AgencyBalance {
  agency_code: string
  agency_name: string
  balance: number
  updated_at: string
}

export interface BillingInfo {
  price_per_million: number
  init_balance: number
  balances: AgencyBalance[]
}

export interface UsageRecord {
  id: number
  username: string
  display_name: string
  feature: string
  model: string
  prompt_tokens: number
  completion_tokens: number
  total_tokens: number
  created_at: string
}

export const getUsageSummary = () =>
  api.get<{ ok: boolean; data: UsageSummaryRow[]; price_per_million: number }>('/llm-usage/summary')

export const getUsageRecent = (limit = 100) =>
  api.get<{ ok: boolean; data: UsageRecord[] }>(`/llm-usage/recent?limit=${limit}`)

export const getBilling = () =>
  api.get<{ ok: boolean; data: BillingInfo }>('/llm-usage/billing')

export const updatePrice = (price: number) =>
  api.put<{ ok: boolean; data: { price_per_million: number } }>('/llm-usage/price', { price })

export const updateBalance = (agency_code: string, balance: number) =>
  api.put<{ ok: boolean; data: { balance: number } }>('/llm-usage/balance', { agency_code, balance })

File: backend/models/llm_usage.py
from . import db


class LlmUsage(db.Model):
    """大模型调用的 token 用量记录（按账号、按功能统计）。"""
    __tablename__ = "llm_usage"

    id = db.Column(db.Integer, primary_key=True)
    username = db.Column(db.String(100), default="", index=True)  # 调用者账号
    display_name = db.Column(db.String(50), default="")            # 调用者姓名
    feature = db.Column(db.String(40), default="", index=True)     # 功能标识，如 ai-review
    model = db.Column(db.String(80), default="")                   # 所用模型名
    prompt_tokens = db.Column(db.Integer, default=0)
    completion_tokens = db.Column(db.Integer, default=0)
    total_tokens = db.Column(db.Integer, default=0)
    created_at = db.Column(db.String(30), default="", index=True)

    def to_dict(self):
        return {
            "id": self.id,
            "username": self.username or "",
            "display_name": self.display_name or "",
            "feature": self.feature or "",
            "model": self.model or "",
            "prompt_tokens": self.prompt_tokens or 0,
            "completion_tokens": self.completion_tokens or 0,
            "total_tokens": self.total_tokens or 0,
            "created_at": self.created_at or "",
        }
